spam/03_spam_data_slicing_tutorial.py

```diff
@@ -0,0 +1,483 @@
+X_valid, Y_valid = df_to_torch_example(vectorizer, df_valid, fit_train=False)
+X_test, Y_test = df_to_torch_example(vectorizer, df_test, fit_train=False)
+
+# %% [markdown]
+# ### Create DataLoaders
+#
+# Next, we'll use the extracted Tensors to initialize a `DictDataLoader` — as a quick recap, this is a Snorkel-specific class that inherits from the common PyTorch class and supports multiple data fields in the `X_dict` and labels in the `Y_dict`.
+#
+# In this task, we'd like to store the `bow_features` in our `X_dict`, and we have one set of labels (for now) correpsonding to the `spam_task`.
+
+# %%
+BATCH_SIZE = 32
+
+
+def create_dict_dataloader(X, Y, split, **kwargs):
+    ds = DictDataset(
+        name="spam_dataset",
+        split=split,
+        X_dict={"bow_features": torch.FloatTensor(X)},
+        Y_dict={"spam_task": torch.LongTensor(Y)},
+    )
+    return DictDataLoader(ds, **kwargs)
+
+
+dl_train = create_dict_dataloader(
+    X_train, Y_train, split="train", batch_size=BATCH_SIZE, shuffle=True
+)
+dl_valid = create_dict_dataloader(
+    X_valid, Y_valid, split="valid", batch_size=BATCH_SIZE, shuffle=False
+)
+dl_test = create_dict_dataloader(
+    X_test, Y_test, split="test", batch_size=BATCH_SIZE, shuffle=False
+)
+
+# %% [markdown]
+# We can inspect our datasets to confirm that they have the appropriate fields.
+
+# %%
+dl_valid.dataset
+
+# %% [markdown]
+# ### Define `SnorkelClassifier`
+#
+# We'll define a simple Multi-Layer Perceptron (MLP) architecture to learn from the `bow_features`.
+#
+# _Note: the following might feel like extra steps to define what is a very simple architecture, but this will lend us additional flexibility later in the pipeline!_
+
+# %% [markdown]
+# To start, we define a `module_pool` with all the [PyTorch](https://pytorch.org) modules that we'll want to include in our network.
+
+# %%
+import torch.nn as nn
+
+bow_dim = X_train.shape[1]
+module_pool = nn.ModuleDict(
+    {
+        "mlp": nn.Sequential(nn.Linear(bow_dim, bow_dim), nn.ReLU()),
+        "prediction_head": nn.Linear(bow_dim, 2),
+    }
+)
+
+# %% [markdown]
+# Then, we specify the desired `task_flow` through each module.
+
+# %%
+from snorkel.classification.task import Operation
+
+task_flow = [
+    Operation(name="input_op", module_name="mlp", inputs=[("_input_", "bow_features")]),
+    Operation(name="head_op", module_name="prediction_head", inputs=[("input_op", 0)]),
+]
+
+# %% [markdown]
+# With these pieces, we're ready to define a [`Task`](https://snorkel.readthedocs.io/en/redux/source/snorkel.classification.html#module-snorkel.classification.task) in Snorkel for spam classification.
+
+# %%
+from functools import partial
+from snorkel.classification.task import Task, ce_loss, softmax
+from snorkel.classification.scorer import Scorer
+
+spam_task = Task(
+    name="spam_task",
+    module_pool=module_pool,
+    task_flow=task_flow,
+    loss_func=partial(ce_loss, "head_op"),
```

Comment:
You can drop loss_func and output_func here, since we use ce_loss and softmax on the last op by default.